wolfssl/wolfcrypt/wc_port.h

```diff
@@ -371,6 +430,54 @@
 
     #endif /* USE_WOLFSSL_LINUXKM_PIE_REDIRECT_TABLE */
 
+#ifdef WOLFSSL_LINUXKM_SIMD
+
+#ifdef WOLFSSL_LINUXKM_SIMD_X86
+
+#ifdef WOLFSSL_LINUXKM_SIMD_X86_IRQ_ALLOWED
+    extern __must_check int allocate_wolfcrypt_irq_fpu_states(void);
+    extern void free_wolfcrypt_irq_fpu_states(void);
+    extern __must_check int save_vector_registers_x86(void);
+    extern void restore_vector_registers_x86(void);
+#else /* !WOLFSSL_LINUXKM_SIMD_X86_IRQ_ALLOWED */
+    static __must_check inline int save_vector_registers_x86(void) {
+        preempt_disable();
+        if (! irq_fpu_usable()) {
+            preempt_enable();
+            return EPERM;
+        } else {
+            kernel_fpu_begin();
+            preempt_enable(); /* kernel_fpu_begin() does its own preempt_disable().  decrement ours. */
+            return 0;
+        }
+    }
+    static inline void restore_vector_registers_x86(void) {
+        kernel_fpu_end();
+    }
+#endif /* !WOLFSSL_LINUXKM_SIMD_X86_IRQ_ALLOWED */
+
+#elif defined(CONFIG_ARM) || defined(CONFIG_ARM64)
+
+    static __must_check inline int save_vector_registers_arm(void) {
+        preempt_disable();
+        if (! may_use_simd()) {
+            preempt_enable();
+            return EPERM;
```

Comment:
Should this return the `BAD_STATE_E`? Not sure our calling functions will be happy with `EPERM`.

Reply:
fixed.